src/components/Quiz.svelte

```diff
@@ -1,44 +1,75 @@
 <script>
+    import Score from './Score.svelte';
     import questions from '../data/questions';
+    import { shuffle } from '../helpers/helpers';
 
     let hasReplied = false;
+    let hasRepliedBonus = false;
     let activeQuestion = 0;
     let isReplyCorrect;
+    let isBonusCorrect;
     let score = 0;
+    let shuffledQuestions = [];
+    let bonusPropositions = [];
+    let bonusCorrectAnswer = -1;
 
     const checkResponse = (from) => {
-        if (from === questions[activeQuestion].answer) {
+        if (from === shuffledQuestions[activeQuestion].answer) {
             isReplyCorrect = true;
             score++;
+            getRandomAnswers(shuffledQuestions[activeQuestion].answer, shuffledQuestions[activeQuestion].bonus);
+        } else {
+            isReplyCorrect = false;
         }
 
         hasReplied = true;
         return;
     };
 
+    const checkBonus = (answer) => {
+        // If user answered correctly
+        if (answer === bonusPropositions[bonusCorrectAnswer].bonus) {
+            score++;
+            isBonusCorrect = true;
+        } else {
+            isBonusCorrect = false;
+        }
+        hasRepliedBonus = true;
+    };
+
+    const getRandomAnswers = (type, excluded) => {
+        bonusPropositions = questions.filter(question => question.answer === type && question.bonus !== excluded);
+        bonusPropositions.sort(() => Math.random() - 0.5);
```

Comment:
Attention avec les mutations en Svelte, ça ne marche pas comme on s'y attend, voir https://svelte.dev/tutorial/updating-arrays-and-objects
Ici bonusPropositions.sort n'a aucun effet.